pkg/controllermanager/controller/project/project_control_reconcile.go

```diff
@@ -251,6 +251,10 @@ func (c *defaultControl) reconcileNamespaceForProject(ctx context.Context, garde
 		ns.Labels = utils.MergeStringMaps(ns.Labels, projectLabels)
 		ns.Annotations = utils.MergeStringMaps(ns.Annotations, projectAnnotations)
 
+		if _, ok := ns.Annotations[common.ProjectCausedNamespaceCreation]; !ok {
```

Comment:
What about namespaces that have been created before this change and thus don't have a `"project.gardener.cloud/caused-namespace-creation"` annotation?

Reply:
Nothing, we cannot handle them because we cannot decide whether they were created by us or by somebody else. I've clarified it in the release note.

Reply:
@timuthy I misunderstood your comment earlier. After a sync with @timebertt I've adapted the logic now.